pkg/reconciler/reconciler.go

```diff
@@ -537,10 +556,11 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 	}
 	u.UpdateStatus(updater.EnsureCondition(conditions.Irreconcilable(corev1.ConditionFalse, "", "")))
 
-	for _, h := range r.preHooks {
-		if err := h.Exec(obj, vals, log); err != nil {
-			log.Error(err, "pre-release hook failed")
-		}
+	r.extensions.loggerInto(log)
+
+	err = r.extPreReconcile(ctx, obj, release, vals)
+	if err != nil {
+		log.Error(err, "pre-release hook failed")
```

Comment:
No return? This means that if an extension fails, we'll not try again after a back-off. This limitation of pre hooks was part of the reason why I introduced extensions in the first place. Again, if hook functions require special retrofitting (ignoring returned errors), this should happen at registration time - meaning your hook wrapper extension is responsible of logging the error, if any, and then unconditionally returning a `nil` error from `PreReconcile`.

Reply:
The return is missing, thanks.

Reply:
It's not just a missing return. What's the point of logging if you're returning the error anyway? Also, if the aim of the hook wrappers is to be backwards-compatible, then you need to move the logging and the `return nil` to the respective methods. Otherwise you would change the behavior.

Reply:
changed, pushed